packages/core/src/Cardano/util/computeImplicitCoin.ts

```diff
@@ -16,15 +18,30 @@ export interface ImplicitCoin {
 
 type DepositProtocolParams = { stakeKeyDeposit: Cardano.Lovelace; poolDeposit: Cardano.Lovelace };
 
+const stakeCredentialInRewardAccounts = (
+  stakeCredential: Cardano.Credential,
+  rewardAccounts: Cardano.RewardAccount[]
+): boolean => {
+  // No reward accounts means accept any stake credential
+  if (rewardAccounts.length === 0) return true;
+  const networkId = Cardano.RewardAccount.toNetworkId(rewardAccounts[0]);
+  return rewardAccounts.includes(Cardano.RewardAccount.fromCredential(stakeCredential, networkId));
+};
+
 const computeShellyDeposits = (
   depositParams: DepositProtocolParams,
   certificates: Cardano.Certificate[],
-  rewardAccounts: Cardano.RewardAccount[],
-  poolIds: Set<Cardano.PoolId>,
-  networkId: Cardano.NetworkId
+  rewardAccounts: Cardano.RewardAccount[]
 ): { deposit: Cardano.Lovelace; reclaimDeposit: Cardano.Lovelace } => {
   let deposit = 0n;
   let reclaimDeposit = 0n;
+  const anyRewardAccount = rewardAccounts.length === 0;
+
+  const poolIds = new Set(
+    rewardAccounts
+      .map((account) => Cardano.RewardAccount.toHash(account))
+      .map((hash) => Cardano.PoolId.fromKeyHash(hash))
```

Comment:
```suggestion
    rewardAccounts.map((account) => Cardano.RewardAccount.toHash(account).Cardano.PoolId.fromKeyHash(hash))
```

Could this work as well? It would use une less `map`.

Reply:
done here [cbc1744](https://github.com/input-output-hk/cardano-js-sdk/pull/1069/commits/cbc174488c54ff0d55d91d5b1b9559569aba5cbc)